jbpm-designer-backend/src/main/java/org/jbpm/designer/web/server/JbpmServiceRepositoryServlet.java

```diff
@@ -227,22 +250,30 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
 		} 
 	}
 
-	private byte[] getImageBytes(InputStream is) throws Exception {
+	private String getFileIconEncoded(String fileIconPath) {
 		try {
-			return IOUtils.toByteArray(is);
-		}
-		catch (IOException e) {
-			throw new Exception("Error creating image byte array.");
-		}
-		finally {
-			if (is != null) { is.close(); }
+			return javax.xml.bind.DatatypeConverter.printBase64Binary (
+					IOUtils.toByteArray( new FileInputStream( new File( fileIconPath.substring(5) ) ) )
+			);
+		} catch(Exception e) {
+			try {
+				// return default service icon
+				String defaultServiceNodeIcon = getServletContext().getRealPath(ConfigurationProvider.getInstance().getDesignerContext() + "/defaults/defaultservicenodeicon.png");
+				return javax.xml.bind.DatatypeConverter.printBase64Binary(
+						IOUtils.toByteArray( new FileInputStream( new File( defaultServiceNodeIcon ) ) )
+				);
+			} catch(Exception ee) {
+				_logger.error("Unable to load workitem icon: " + ee.getMessage());
+				return "";
+			}
 		}
 	}
 
-    private String getRepositoryDir(String uuid) {
-        int iStart = uuid.indexOf("//");
-        iStart = uuid.indexOf('/', iStart + 2);
-        int iEnd = uuid.lastIndexOf('/');
-        return uuid.substring(iStart, iEnd);
-    }
+	public boolean canUpdateConfigForWorkitem(WorkDefinitionImpl workitem) {
```

Comment:
Please remove this method. Its copy is present and used in ServiceRepoUtils class.